lib/url.c

```diff
@@ -2542,6 +2551,30 @@ static CURLcode parse_proxy(struct Curl_easy *data,
     result = CURLE_OUT_OF_MEMORY;
     goto error;
   }
+#ifdef USE_UNIX_SOCKETS
+  if(sockstype && strcmp(UNIX_SOCKET_PREFIX, host) == 0) {
+    uc = curl_url_get(uhp, CURLUPART_PATH, &path, CURLU_URLDECODE);
+    if(uc) {
+      result = CURLE_OUT_OF_MEMORY;
+      goto error;
+    }
+    /* path will be "/", if no path was was found */
+    if(!(path[0]=='/' && path[1]=='\0')) {
+      is_unix_proxy = TRUE;
+      Curl_safefree(host);
+      host = aprintf(UNIX_SOCKET_PREFIX"%s", path);
+      if(!host) {
+        result = CURLE_OUT_OF_MEMORY;
+        goto error;
+      }
+      Curl_safefree(proxyinfo->host.rawalloc);
+      proxyinfo->host.rawalloc = host;
+      proxyinfo->host.name = host;
+    }
+  }
+
+  if(!is_unix_proxy) {
+#endif
   Curl_safefree(proxyinfo->host.rawalloc);
```

Comment:
This line should be indented to the right here, right?

Reply:
Fixed